Mathlib/Topology/Algebra/Order/Compact.lean

```diff
@@ -193,6 +193,73 @@ theorem IsCompact.exists_isLUB [ClosedIciTopology α] {s : Set α} (hs : IsCompa
   IsCompact.exists_isGLB (α := αᵒᵈ) hs ne_s
 #align is_compact.exists_is_lub IsCompact.exists_isLUB
 
+theorem _root_.cocompact_le_atBot_atTop [LinearOrder α] [CompactIccSpace α] :
+    cocompact α ≤ atBot ⊔ atTop := by
+  refine fun s hs ↦ mem_cocompact.mpr <| (isEmpty_or_nonempty α).casesOn ?_ ?_ <;> intro
+  · exact ⟨∅, isCompact_empty, fun x _ ↦ (IsEmpty.false x).elim⟩
+  · obtain ⟨t, ht⟩ := mem_atBot_sets.mp hs.1
+    obtain ⟨u, hu⟩ := mem_atTop_sets.mp hs.2
+    refine ⟨Icc t u, isCompact_Icc, fun x hx ↦ ?_⟩
+    exact (not_and_or.mp hx).casesOn (fun h ↦ ht x (le_of_not_le h)) fun h ↦ hu x (le_of_not_le h)
+
+theorem _root_.cocompact_le_atBot [LinearOrder α] [OrderTop α] [CompactIccSpace α] :
+    cocompact α ≤ atBot := by
+  refine fun _ hs ↦ mem_cocompact.mpr <| (isEmpty_or_nonempty α).casesOn ?_ ?_ <;> intro
+  · exact ⟨∅, isCompact_empty, fun x _ ↦ (IsEmpty.false x).elim⟩
+  · obtain ⟨t, ht⟩ := mem_atBot_sets.mp hs
+    refine ⟨Icc t ⊤, isCompact_Icc, fun _ hx ↦ ?_⟩
+    exact (not_and_or.mp hx).casesOn (fun h ↦ ht _ (le_of_not_le h)) (fun h ↦ (h le_top).elim)
+
+theorem _root_.cocompact_le_atTop [LinearOrder α] [OrderBot α] [CompactIccSpace α] :
+    cocompact α ≤ atTop := by
+  refine fun _ hs ↦ mem_cocompact.mpr <| (isEmpty_or_nonempty α).casesOn ?_ ?_ <;> intro
```

Comment:
No need to do twice the same proof. If you register an instance
```
instance [CompactIccSpace α] : CompactIccSpace (αᵒᵈ) := sorry
```
(close to the definition of the `CompactIccSpace` class), then you can prove this theorem as
`:= cocompact_le_atBot (α := αᵒᵈ)`.

Reply:
Thanks!